criu/pidfd.c

```diff
@@ -199,6 +207,18 @@ static int open_one_pidfd(struct file_desc *d, int *new_fd)
 	BUG_ON(dead->count == 0);
 	dead->count--;
 	if (dead->pid == -1) {
+		/* Block SIGCHLD to prevent interfering from sigchld_handler()
+		 * and to properly handle the tmp process termination without
+		 * a race condition. A similar approach is used in cr_system().
+		 */
+		sigemptyset(&blockmask);
+		sigaddset(&blockmask, SIGCHLD);
+		if (sigprocmask(SIG_BLOCK, &blockmask, &dead->oldmask) == -1) {
```

Comment:
sigprocmask should be called right before killing this process. If I remember it right, a process can be created in one process, but killed in another process.

Reply:
It makes sense. I've updated the pull request.